docs/src/content/docs/reference/plugins.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Plugins Reference
+description: An overview of the Starlight plugin API.
+---
+
+Starlight Plugins provide an API to customize Starlight configuration, UI and behavior while also being easy to share and reuse.
+This page is for anyone interested in creating their own Starlight plugins.
+
+Lean more about using a Starlight plugin in the [Configuration Reference](/reference/configuration/#plugins) or visit the [Starlight Plugins Showcase](/showcase/#community-plugins) to see a list of community plugins.
+
+## Quick API Reference
+
+```ts
+interface StarlightPlugin {
+  name: string;
+  plugin: (options: {
+    config: StarlightUserConfig;
+    updateConfig: (newConfig: StarlightUserConfig) => void;
+    addIntegration: (integration: AstroIntegration) => void;
+    logger: AstroIntegrationLogger;
+  }) => void | Promise<void>;
+}
```

Comment:
I was never a huge fan of the `plugin()` method name but did not managed at the time to come up with a better name I liked. I personally like `setup()` so I'm all for it.

Regarding the structure, I personally do not have a strong preference, so I think if you have one, we should go with it otherwise I guess mirroring Astro's structure may be the safest bet?

Reply:
No strong preference from me, so yeah, +1 to matching Astro’s shape 👍